cmd/juju/cloud/addcredential.go

```diff
@@ -392,47 +309,50 @@ func (c *addCredentialCommand) promptCredentialAttributes(
 }
 
 func (c *addCredentialCommand) promptFieldValue(
-	out io.Writer, in io.Reader, attr jujucloud.NamedCredentialAttr,
+	p *interact.Pollster, attr jujucloud.NamedCredentialAttr,
 ) (string, error) {
-
 	name := attr.Name
-	// Formulate the prompt for the list of valid options.
-	optionsPrompt := ""
+
 	if len(attr.Options) > 0 {
 		options := make([]string, len(attr.Options))
 		for i, opt := range attr.Options {
 			options[i] = fmt.Sprintf("%v", opt)
-			if i == 0 {
-				options[i] += "*"
-			}
 		}
-		optionsPrompt = fmt.Sprintf(" [%v]", strings.Join(options, ","))
-	}
-
-	// Prompt for and accept input for field value.
-	fmt.Fprintf(out, "Enter %s%s: ", name, optionsPrompt)
-	var input string
-	var err error
-	if attr.Hidden {
-		input, err = c.readHiddenField(in)
-		fmt.Fprintln(out)
-	} else {
-		input, err = readLine(in)
-	}
-	if err != nil {
-		return "", errors.Trace(err)
+		return p.Select(interact.List{
+			Singular: name,
+			Plural:   name,
+			Options:  options,
+			Default:  options[0],
+		})
+	}
+
+	// We assume that Hidden and FilePath are mutually exclusive here.
+	switch {
+	case attr.Hidden:
+		return p.EnterPassword(name)
+	case attr.FilePath:
+		return enterFile(name, p)
+	case attr.Optional:
+		return p.EnterOptional(name)
+	default:
+		return p.Enter(name)
 	}
-	value := strings.TrimSpace(input)
-	return value, nil
 }
 
-func (c *addCredentialCommand) readHiddenField(in io.Reader) (string, error) {
-	if f, ok := in.(*os.File); ok && terminal.IsTerminal(int(f.Fd())) {
-		value, err := terminal.ReadPassword(int(f.Fd()))
+func enterFile(name string, p *interact.Pollster) (string, error) {
```

Comment:
Seems like prompting for a file might be a common enough case to add to pollster. Maybe.